File: Treachery-iOS/Treachery-iOS/Models/PlaneCard.swift
//
//  PlaneCard.swift
//  Treachery-iOS
//
//  Created by Luke Solomon on 3/16/26.
//

import Foundation

struct PlaneCard: Codable, Identifiable {
    let id: String
    let name: String
    let typeLine: String
    let oracleText: String
    let imageUri: String?
    let isPhenomenon: Bool

    enum CodingKeys: String, CodingKey {
        case id
        case name
        case typeLine = "type_line"
        case oracleText = "oracle_text"
        case imageUri = "image_uri"
        case isPhenomenon = "is_phenomenon"
    }
}

File: TreacheryAndroid/app/src/main/java/com/solomon/treachery/model/PlaneCard.kt
package com.solomon.treachery.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class PlaneCard(
    val id: String,
    val name: String,
    @SerialName("type_line") val typeLine: String,
    @SerialName("oracle_text") val oracleText: String,
    @SerialName("image_uri") val imageUri: String? = null,
    @SerialName("is_phenomenon") val isPhenomenon: Boolean
)
